[website] Update cli-tools document (#11980)
apache/pulsar: site2/docs/reference-cli-tools.md

|`-s`, `--separator`|Character to split messages string with.|","|
|`-k`, `--key`|Message key to add|key=value string, like k1=v1,k2=v2.|
|`-p`, `--properties`|Properties to add. If you want to add multiple properties, use the comma as the separator, e.g. `k1=v1,k2=v2`.| |
|`-ekn`, `--encryption-key-name`|The public key name to encrypt payload.| |

> |`-ekv`, `--encryption-key-value`|The URI of public key to encrypt payload, for example file:///path/to/public.key or data:application/x-pem-file;base64,*****.| |

Reviewer: ```suggestion
|`-ekv`, `--encryption-key-value`|The URI of public key to encrypt payload. For example, `file:///path/to/public.key or data:application/x-pem-file;base64,*****`.| |
```

|`-ac`, `--auto_ack_chunk_q_full`|Auto ack for the oldest message in consumer's receiver queue if the queue full.|false|
|`--hide-content`|Do not print the message to the console.|false|
|`-st`, `--schema-type`|Set the schema type. Use `auto_consume` to dump AVRO and other structured data types. Possible values: bytes, auto_consume.|bytes|

> |`-ekv`, `--encryption-key-value`|The URI of public key to encrypt payload, for example file:///path/to/public.key or data:application/x-pem-file;base64,*****.| |

Reviewer: same

|Flag|Description|Default|
|---|---|---|

> |`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||

Reviewer: same comment as previously

|`--auth_plugin`|Authentication plugin class name||
|`-ac`, `--auto_ack_chunk_q_full`|Auto ack for oldest message on queue is full|false|
|`--listener-name`|Listener name for the broker||
|`--acks-delay-millis`|Acknowledgements grouping delay in millis|100|
|`--batch-index-ack`|Enable or disable the batch index acknowledgment|false|
|`-bw`, `--busy-wait`|Enable Busy-Wait on the Pulsar client|false|
|`-v`, `--encryption-key-value-file`|The file which contains the private key to decrypt payload||
|`-h`, `--help`|Help message|false|
|`--conf-file`|Configuration file||

> |`-e`, `--expire_time_incomplete_chunked_messages`|Expire time in ms for incomplete chunk messages|0|

Reviewer: same comment as previously

|`-n`, `--num-consumers`|Number of consumers (per topic)|1|
|`-ioThreads`, `--num-io-threads`|Set the number of threads to be used for handling connections to brokers|1|
|`-ns`, `--num-subscriptions`|Number of subscriptions (per topic)|1|
|`-t`, `--num-topics`|The number of topics|1|
|`-pm`, `--pool-messages`|Use the pooled message|true|
|`-r`, `--rate`|Simulate a slow message consumer (rate in msg/s)|0|
|`-q`, `--receiver-queue-size`|Size of the receiver queue|1000|
|`-p`, `--receiver-queue-size-across-partitions`|Max total size of the receiver queue across partitions|50000|
|`--replicated`|Whether the subscription status should be replicated|false|
|`-u`, `--service-url`|Pulsar service URL||
|`-i`, `--stats-interval-seconds`|Statistics interval seconds. If 0, statistics will be disabled|0|
|`-s`, `--subscriber-name`|Subscriber name prefix||
|`-ss`, `--subscriptions`|A list of subscriptions to consume on (e.g. sub1,sub2)|sub|
|`-st`, `--subscription-type`|Subscriber type. Possible values are Exclusive, Shared, Failover, Key_Shared.|Exclusive|
|`-sp`, `--subscription-position`|Subscriber position. Possible values are Latest, Earliest.|Latest|

> |`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep consuming|0|

Reviewer: same comment as previously

|Flag|Description|Default|
|---|---|---|

> |`-am`, `--access-mode`|Producer access mode, valid values: [Shared, Exclusive, WaitForExclusive]|Shared|

Reviewer: same comment as previously

|`-au`, `--admin-url`|Pulsar Admin URL||

> |`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||

Reviewer: same comment as previously

|`--auth_plugin`|Authentication plugin class name||
|`--listener-name`|Listener name for the broker||
|`-b`, `--batch-time-window`|Batch messages in a window of the specified number of milliseconds|1|
|`-bb`, `--batch-max-bytes`|Maximum number of bytes per batch|4194304|
|`-bm`, `--batch-max-messages`|Maximum number of messages per batch|1000|
|`-bw`, `--busy-wait`|Enable Busy-Wait on the Pulsar client|false|

> |`-ch`, `--chunking`|Should split the message and publish in chunks if message size is larger than allowed max size|false|

Reviewer: ```suggestion
|`-ch`, `--chunking`|Split the message and publish in chunks if the message size is larger than allowed max size|false|
```

|`-d`, `--delay`|Mark messages with a given delay in seconds|0s|
|`-z`, `--compression`|Compress messages’ payload. Possible values are NONE, LZ4, ZLIB, ZSTD or SNAPPY.||
|`--conf-file`|Configuration file||
|`-k`, `--encryption-key-name`|The public key name to encrypt payload||
|`-v`, `--encryption-key-value-file`|The file which contains the public key to encrypt payload||
|`-ef`, `--exit-on-failure`|Exit from the process on publish failure|false|
|`-fc`, `--format-class`|Custom Formatter class name|org.apache.pulsar.testclient.DefaultMessageFormatter|
|`-fp`, `--format-payload`|Format %i as a message index in the stream from producer and/or %t as the timestamp nanoseconds|false|
|`-h`, `--help`|Help message|false|
|`-c`, `--max-connections`|Max number of TCP connections to a single broker|100|
|`-o`, `--max-outstanding`|Max number of outstanding messages|1000|
|`-p`, `--max-outstanding-across-partitions`|Max number of outstanding messages across partitions|50000|

> |`-mk`, `--message-key-generation-mode`|The generation mode of message key, valid options: [autoIncrement, random]||

Reviewer: same comment as previously

|`-ioThreads`, `--num-io-threads`|Set the number of threads to be used for handling connections to brokers|1|

> |`-m`, `--num-messages`|Number of messages to publish in total. If <= 0, it will keep publishing.|0|

Reviewer: same comment as previously

|`-n`, `--num-producers`|The number of producers (per topic)|1|
|`-threads`, `--num-test-threads`|Number of test threads|1|
|`-t`, `--num-topic`|The number of topics|1|

> |`-np`, `--partitions`|Create partitioned topics with the given number of partitions, set 0 to not try to create the topic||

Reviewer: ```suggestion
|`-np`, `--partitions`|Create partitioned topics with the given number of partitions. Setting this value to 0 means not trying to create a topic||
```

|`-f`, `--payload-file`|Use payload from an UTF-8 encoded text file and a payload will be randomly selected when publishing messages||
|`-e`, `--payload-delimiter`|The delimiter used to split lines when using payload from a file|\n|
|`-pn`, `--producer-name`|Producer Name||
|`-r`, `--rate`|Publish rate msg/s across topics|100|
|`--send-timeout`|Set the sendTimeout|0|
|`--separator`|Separator between the topic and topic number|-|
|`-u`, `--service-url`|Pulsar service URL||
|`-s`, `--size`|Message size (in bytes)|1024|
|`-i`, `--stats-interval-seconds`|Statistics interval seconds. If 0, statistics will be disabled.|0|

> |`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep publishing.|0|

Reviewer: same comment as previously

```

Options

|Flag|Description|Default|
|---|---|---|

> |`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||

Reviewer: same comment as previously

|`--auth_plugin`|Authentication plugin class name||
|`--listener-name`|Listener name for the broker||
|`--conf-file`|Configuration file||
|`-h`, `--help`|Help message|false|
|`-c`, `--max-connections`|Max number of TCP connections to a single broker|100|
|`-ioThreads`, `--num-io-threads`|Set the number of threads to be used for handling connections to brokers|1|
|`-t`, `--num-topics`|The number of topics|1|
|`-r`, `--rate`|Simulate a slow message reader (rate in msg/s)|0|
|`-q`, `--receiver-queue-size`|Size of the receiver queue|1000|
|`-u`, `--service-url`|Pulsar service URL||
|`-m`, `--start-message-id`|Start message id. This can be either 'earliest', 'latest' or a specific message id by using 'lid:eid'|earliest|
|`-i`, `--stats-interval-seconds`|Statistics interval seconds. If 0, statistics will be disabled.|0|

> |`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep consuming|0|

Reviewer: same comment as previously

```

Options

|Flag|Description|Default|
|---|---|---|

> |`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||

Reviewer: same comment as previously

|`--auth_plugin`|Authentication plugin class name||
|`--conf-file`|Configuration file||
|`-h`, `--help`|Help message|false|

> |`-m`, `--num-messages`|Number of messages to publish in total. If <= 0, it will keep publishing|0|

Reviewer: same comment as previously

|`-t`, `--num-topic`|The number of topics|1|
|`-f`, `--payload-file`|Use payload from a file instead of empty buffer||
|`-u`, `--proxy-url`|Pulsar Proxy URL, e.g., "ws://localhost:8080/"||
|`-r`, `--rate`|Publish rate msg/s across topics|100|
|`-s`, `--size`|Message size in byte|1024|

> |`-time`, `--test-duration`|Test duration in secs. If <=0, it will keep publishing|0|

Reviewer: same comment as previously

|`-h`, `--help`|Help message|false|
|`-c`, `--max-connections`|Max number of TCP connections to a single bookie|1|
|`-o`, `--max-outstanding`|Max number of outstanding requests|1000|

> |`-m`, `--num-messages`|Number of messages to publish in total. If <= 0, it will keep publishing|0|

Reviewer: same comment as previously

|`-t`, `--num-topic`|Number of managed ledgers|1|
|`-r`, `--rate`|Write rate msg/s across managed ledgers|100|
|`-s`, `--size`|Message size in byte|1024|

> |`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep publishing|0|

Reviewer: same comment as previously
